### DISCUSSION TITLE
Does milvus2.x's Coord support HA? How is it implemented? What is the solution used?

### DISCUSSION THREAD

**Discussion by ponponon** [milvus-io/milvus#23157] [Q&A and General discussion]
milvus2.x 的 Coord 支持 HA 吗？怎么实现的？用的什么方案？

Does milvus2.x's Coord support HA? How is it implemented? What is the solution used?

from: https://milvus.io/docs/release_notes.md#Features

貌似是主备？为什么要用主备而不是副本？这套主备是怎么工作的？master 工作，master 挂了，slave 顶上去吗？master 工作的时候，slave 只能闲置看戏？如果 master 挂了，slave 要多久才能顶上去? （意思就是如何实现 slave 发现 master 挂了的，并且提升自己？）要多久？一纳秒？一毫秒？一秒钟？或者一分钟 slave 才能顶上去吗？

It seems to be a master backup? Why do we need a master instead of a replica? How does the master backup work, does the master work, does the slave take over when the master is down, and does the slave just sit around and watch while the master works? How long does it take for a slave to take over if the master goes down? (meaning how does the slave find out that the master is down and raise itself?) How long does it take? A nanosecond? A millisecond? A second? Or a minute for the slave to take over?

**Comment by xiaofan-luan** [comment]
Usually you don't really need HA in K8s because it helps to automatically bring a new coordinator.
For user really care about availability or without K8s, you can bring up two coordinators. one of coordinator will be standby and takeover once the previous leader etcd session expire
